### ISSUE TITLE
`torch.nn.functional.l1_loss` is "wrong" for complex inputs

### ISSUE THREAD

**Issue by pmeier** [pytorch/pytorch#65684]
Although there is no definition of L norms for complex inputs (or at least I didn't find any), our current implementation of the L1 norm is probably not doing what a user would expect:

https://github.com/pytorch/pytorch/blob/facff2ec6562d61eab2ccd9f1c16872e54eec92d/aten/src/ATen/native/Loss.cpp#L499

`abs(diff)` is fine for real inputs, but for complex inputs this is equivalent to `sqrt((input.real - target.real) ** 2 + (input.imag - target.imag) ** 2)`. This closely resembles the L2 norm rather the L1 norm.

A version probably more aligned with the user expectation, would be to perform the `abs(diff)` componentwise and sum up the results.

cc @albanD @mruberry @jbschlosser @walterddr @ezyang @anjali411 @dylanbespalko @Lezcano @nikitaved

**Comment by nikitaved** [comment]
The definition of L_p norm is defined for both reals and complex, the L1 norm is just the sum of absolute values.

**Comment by Lezcano** [comment]
Absolute values component-wise, right? That is, \norm{x+iy}_1 = \abs{x} + \abs{y} @nikitaved

**Comment by Lezcano** [comment]
Nevermind, @nikitaved was right, the L_p norm on C^n is defined as the  L_p norm of the real vector of L_2 norms of the components.
